boututils/datafile.py

```diff
@@ -473,14 +473,32 @@ def _bout_type_from_dimensions(self, varname):
         dims_dict = {
             ('t', 'x', 'y', 'z'): "Field3D_t",
             ('t', 'x', 'y'): "Field2D_t",
+            ('t', 'x', 'z'): "FieldPerp_t",
             ('t',): "scalar_t",
             ('x', 'y', 'z'): "Field3D",
             ('x', 'y'): "Field2D",
+            ('x', 'z'): "FieldPerp",
+            ('x'): "ArrayX",
             (): "scalar",
         }
 
         return dims_dict.get(dims, None)
 
+    def _bout_dimensions_from_type(self, bout_type):
+        dims_dict = {
+            "Field3D_t": ('t', 'x', 'y', 'z'),
+            "Field2D_t": ('t', 'x', 'y'),
+            "FieldPerp_t": ('t', 'x', 'z'),
+            "scalar_t": ('t',),
+            "Field3D": ('x', 'y', 'z'),
+            "Field2D": ('x', 'y'),
+            "FieldPerp": ('x', 'z'),
+            "ArrayX": ('x'),
+            "scalar": (),
```

Comment:
```suggestion
            "scalar": tuple(),
```

Reply:
`()` does create an empty tuple, so I'm going to leave as is, unless anyone has strong feelings that `tuple()` is better style?